runtime/lua/provider/ruby/health.lua

```diff
@@ -0,0 +1,69 @@
+local health = vim.health
+local iswin = vim.loop.os_uname().sysname == 'Windows_NT'
+
+local M = {}
+
+function M.check()
+  health.start('Ruby provider (optional)')
+
+  if health.provider_disabled('ruby') then
+    return
+  end
+
+  if vim.fn.executable('ruby') == 0 or vim.fn.executable('gem') == 0 then
+    health.warn(
+      '`ruby` and `gem` must be in $PATH.',
+      'Install Ruby and verify that `ruby` and `gem` commands work.'
+    )
+    return
+  end
+  health.info('Ruby: ' .. health.system({ 'ruby', '-v' }))
```

Comment:
Unsure if `health.system` is needed here over `vim.fn.system` or `health.cmd_ok`. No real preference, but mentioning in case it got unnoticed.